air/src/air/context.rs

```diff
@@ -115,15 +115,6 @@ impl<B: StarkField, P> AirContext<B, P> {
                 num_aux_assertions > 0,
                 "at least one assertion must be specified against the auxiliary trace segment"
             );
-        } else {
-            assert!(
-                aux_transition_constraint_degrees.is_empty(),
-                "auxiliary transition constraint degrees specified for a single-segment trace"
-            );
-            assert!(
-                num_aux_assertions == 0,
-                "auxiliary assertions specified for a single-segment trace"
-            );
```

Comment:
I think we still may need this check - just under a slightly different set of conditions. Basically, it could be like so:
```rust
if trace_info.is_multi_segment() {
    if !trace_info.logup_gkr_enabled() {
        // make sure at least one aux constraint/assertion is specified
    }
} else {
    // make sure there are no aux constraints/assertions specified
}
```

Reply:
Indeed, fixed it now